src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,120 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int nextEmptyPosition;
+    private T[] elements;
+
+    public ArrayList() {
+        this.nextEmptyPosition = 0;
+        this.elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        checkAndIncreaseCapacity();
+        elements[nextEmptyPosition++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > nextEmptyPosition) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        checkAndIncreaseCapacity();
+        shiftToTheRightFrom(index);
+        nextEmptyPosition++;
+        elements[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (list.size() > elements.length - nextEmptyPosition - 1) {
+            increaseCapacity();
+        }
```

Comment:
method add() will check and increase capacity if need.

Reply:
I thought if the list is too big it would be better to increase capacity firstly and then just copy. Was I wrong?